lib/kennel.rb

```diff
@@ -54,48 +54,88 @@ class << self
   self.err = $stderr
 
   class Engine
-    attr_accessor :strict_imports
+    def initialize(
+      update_generated: true,
+      show_plan: false,
+      require_confirm: nil,
+      update_datadog: false,
+      strict_imports: true
+    )
+      @update_generated = update_generated
+      @show_plan = show_plan
+      @require_confirm =
+        if require_confirm.nil?
+          default_require_confirm?
+        else
+          require_confirm
+        end
+      @update_datadog = update_datadog
+      @strict_imports = strict_imports
+    end
+
+    def run
+      if !update_generated? && !show_plan? && !update_datadog?
+        generated
+        return
+      end
+
+      if show_plan? || update_datadog?
+        # start generation and download in parallel to make planning faster
+        Utils.parallel([:generated, :definitions]) { |m| send m, plain: true }
+      end
+
+      if update_generated?
+        PartsSerializer.new(filter: filter).write(generated)
+      end
+
+      if show_plan? || update_datadog?
```

Comment:
store into a variable and reuse from above
```suggestion
      if needs_syncing
```

Reply:
Done. It's probably a taste thing; I don't see the benefit.